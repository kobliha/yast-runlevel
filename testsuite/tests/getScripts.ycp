/**
 * YaST2 test case for runlevel editor
 *
 * tests function getScripts
 */
{
    include "runlevel/services.ycp";
    include "testsuite.ycp";
    
    map READ = $[
        "init" : $[
            "scripts" : $[
                "runlevels" : $[
                    "aaa" : $[
                        "start" : [ "0", "1", "3" ],
                        "stop" : [ "4", "5", "6" ],
                        ],
                    "bbb" : $[
                        "start" : [ "0", "1", "2" ],
                        "stop" : [ "5", "6" ],
                        ],
                    ],
                "comments" : $[
                    "aaa" : $[
                        "defstart" : [ "0", "1", "3" ],
                        "defstop" : [ "4", "5", "6" ],
                        ],
                    "bbb" : $[
                        "defstart" : [ "0", "1", "2" ],
                        "defstop" : [ "5", "6" ],
                        ],
                    ],
                ]
            ]
        ];
    

    map start_variables_all = $[];
    map services = $[];
    list service_list = [];

    TEST (``(getScripts ()), [READ] , 0);
    DUMP (services);

    READ = $[
        "init" : $[
            "scripts" : $[
                "runlevels" : $[
                    "aaa" : $[
                        "start" : [ "0", "1", "3" ],
                        "stop" : [ "4", "5", "6" ],
                        ],
                    "ccc" : $[
                        "start" : [ "0", "1", "2" ],
                        "stop" : [ "5", "6" ],
                        ],
                    ],
                "comments" : $[
                    "aaa" : $[
                        "defstart" : [ "0", "1", "3" ],
                        "defstop" : [ "4", "5", "6" ],
                        ],
                    "bbb" : $[
                        "defstart" : [ "0", "1", "2" ],
                        "defstop" : [ "5", "6" ],
                        ],
                    ],
                ]
            ]
        ];

    TEST (``(getScripts ()), [READ] , 0);
    DUMP (services);

    
}
