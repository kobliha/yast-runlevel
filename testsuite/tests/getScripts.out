Read	.init.scripts.runlevels $["aaa":$["start":["0", "1", "3"], "stop":["4", "5", "6"]], "bbb":$["start":["0", "1", "2"], "stop":["5", "6"]]]
Read	.init.scripts.comments $["aaa":$["defstart":["0", "1", "3"], "defstop":["4", "5", "6"]], "bbb":$["defstart":["0", "1", "2"], "defstop":["5", "6"]]]
Return	nil
Dump	$["aaa":$["START":true, "defstart":["0", "1", "3"], "defstop":["4", "5", "6"], "start":["0", "1", "3"], "stop":["4", "5", "6"]], "bbb":$["START":true, "defstart":["0", "1", "2"], "defstop":["5", "6"], "start":["0", "1", "2"], "stop":["5", "6"]]]
Read	.init.scripts.runlevels $["aaa":$["start":["0", "1", "3"], "stop":["4", "5", "6"]], "ccc":$["start":["0", "1", "2"], "stop":["5", "6"]]]
Read	.init.scripts.comments $["aaa":$["defstart":["0", "1", "3"], "defstop":["4", "5", "6"]], "bbb":$["defstart":["0", "1", "2"], "defstop":["5", "6"]]]
Return	nil
Dump	$["aaa":$["START":true, "defstart":["0", "1", "3"], "defstop":["4", "5", "6"], "start":["0", "1", "3"], "stop":["4", "5", "6"]], "bbb":$["START":true, "defstart":["0", "1", "2"], "defstop":["5", "6"], "start":[], "stop":[]]]
